File: exercise/week04/php/src/Construct/BackdoorPassUpdater.php
<?php

namespace Construct;

class BackdoorPassUpdater extends NormalArtefactUpdater
{
    public function update(Artefact $artefact): void
    {
        $this->increase($artefact, 1);
        if ($artefact->timeToLive < 11) $this->increase($artefact, 1);
        if ($artefact->timeToLive < 6) $this->increase($artefact, 1);

        $artefact->timeToLive--;

        if ($artefact->timeToLive < 0) {
            $artefact->integrity = 0;
        }
    }
}

File: exercise/week04/java/src/main/java/construct/BackdoorPassUpdater.java
package construct;

public class BackdoorPassUpdater extends NormalArtefactUpdater {
    
    @Override
    public void update(Artefact artefact) {
        increase(artefact, 1);
        if (artefact.timeToLive < 11) increase(artefact, 1);
        if (artefact.timeToLive < 6) increase(artefact, 1);

        artefact.timeToLive--;

        if (artefact.timeToLive < 0) {
            artefact.integrity = 0;
        }
    }
}